src/transformers/modeling_utils.py

```diff
@@ -3560,6 +3549,14 @@ def from_pretrained(
                 )
                 pass
 
+        if (
+            quantization_config is not None
+            and quantization_config.quant_method == QuantizationMethod.AWQ
+            and quantization_config.do_fuse
+        ):
+            model = fuse_awq_modules(model, quantization_config)
+            model._awq_is_fused = True
+
```

Comment:
I see that you are using the quantization_config that was passed in the `from_pretrained`. However, if the fused parameters was stored in the `config.quantization_config`, we won't be able to perform the fusing. This is why I decided to overwrite the `config.quantization_config` using` get_loading_attributes()`. What could be done here is use config.quantization_config instead. LMK what you think.

Reply:
Yes, we should always fuse if the user passes do_fuse or any args that are in `get_loading_attributes()`. Thanks to `get_loading_attributes()`, we overwrite `config.quantization_config` to have information about `do_fuse` or other args. 

What I wanted is to use only one config for quantization. Right now, we have the configuration that was passed and the config that was stored. The merging of the two configurations through `get_loading_attributes()` was motivated by the fact that some args should not be modified such as the `backend` and the `version`.